awswrangler/athena/_write_iceberg.py

```diff
@@ -294,6 +306,10 @@ def to_iceberg(
             "Either path or workgroup path must be specified to store the temporary results."
         )
 
+    glue_table_settings: Union[GlueTableSettings, None] = None
+    if columns_comments is not None:
+        glue_table_settings = GlueTableSettings(columns_comments=columns_comments)
+
```

Comment:
If the argument is changed to be `glue_table_settings`, you can get column comments out of it like this:
```python
glue_table_settings = cast(
    GlueTableSettings,
    glue_table_settings if glue_table_settings else {},
)

columns_comments = glue_table_settings.get("columns_comments")
```